pkg/dyninst/decode/types.go

```diff
@@ -1571,7 +1571,7 @@ func (s *goSliceHeaderType) encodeValueFields(
 				tokenNotCapturedReasonUnavailable,
 			)
 		}
-		sliceLength = int(len(sliceData)) / elementSize
+		sliceLength = min(int(len(sliceData))/elementSize, int(length))
```

Comment:
comment what the deal is here? Say that we might have captured less than `length` for multiple reasons, including because the slice was aliased...